dvc/remote/ssh/__init__.py

```diff
@@ -32,6 +32,9 @@ class RemoteSSH(RemoteBASE):
     PARAM_CHECKSUM = "md5"
     DEFAULT_PORT = 22
     TIMEOUT = 1800
+    # At any given time some of the connections will go over network and
+    # paramiko stuff, so we raise this default.
+    CHECKSUM_JOBS = 8
```

Comment:
So maybe 
```suggestion
    CHECKSUM_JOBS = JOBS
```
?

Reply:
If you want this be set to 4, then this is what you should do). I don't see a reason to tie up these two.

Reply:
should we actually make it a formula that depends on CPU count? if someone has 64 cores why should we sacrifice the speed? :)

Reply:
on the hand, 8 can too extreme if number of cpu is really low

Reply:
The thing is we want more jobs than CPUs, especially for smaller files. I tested locally with 4 hyperthreading cpus and 8 jobs is considerably faster than 4. A user can specify that in config as `checksum_jobs` property.

We may take into account CPU number to make it, but we need to get that one first via ssh. This will also change if/once we implement batching. We may also try 
```bash
xargs -P`nproc` -n1 md5sum < pass-filenames-as-input
```

Reply:
This was about future, returning to this one we should merge it with either 4 or 8. And should add that `checksum_jobs` config option to docs.

Reply:
ah, got it. It's about running remotely, not locally. `xargs` stuff looks complicated. I'm fine to use some constant in this case. Let's probably update docs where appropriate to mention that it's 8 for running staff remotely.

Reply:
**P.S.** We might also show `checksum_jobs` advice when people are adding new ssh remote.

Reply:
> Let's probably update docs where appropriate to mention that it's 8 for running staff remotely.
There is a comment just above it) And in docs it should be obvious anyway - option applies to remote configured whether it be remote or local.

Reply:
I mean that in this doc https://dvc.org/doc/commands-reference/config we explicitly mention the formula we calculate the default value with. It's not correct in this case.

Reply:
I'm talking about JOBS because it determines a number of ssh connections, and 8 might be too much, that is why we are using 4 for ssh remote.

Reply:
DVC's ssh connection pool doesn't deal with the connection limit control currently, right?

Reply:
@ei-grad no, it doesn't. And if it were we will need to modify some threading code probably too, like one in `_collect_dir()`.